src/main/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepo.scala

```diff
@@ -99,16 +94,38 @@ class RoutesPostgresRepo @Inject() (
   override def selectByName(name: String): DatabasePublisher[RouteRow] = {
     logger.debug(s"selectByName $name")
 
-    val q = for {
+    val q = (for {
       routeRow <- routesTable
       if routeRow.name === name && routeRow.deletedAt.isEmpty
-    } yield routeRow
+    } yield routeRow)
 
     db.stream {
       q.result
     }
   }
 
+  override def selectLatestActiveRoutesPerName(currentTime: LocalDateTime): DatabasePublisher[RouteRow] = {
+    logger.debug("select latest routes per name")
+
+    val q = (for {
+      routeRow <- routesTable
+      if routeRow.deletedAt.isEmpty && routeRow.activateAt < currentTime
+    } yield (routeRow.id, routeRow.name)).groupBy(_._2)
+
+    val maxIdsQuery = q.map {
+      case (name, query) =>
```

Comment:
for some reason, its type actually is: `Query[(Rep[Long], Rep[String]), (Long, String), Seq]`